src/platform/Tizen/DnssdImpl.cpp

```diff
@@ -377,14 +372,37 @@ void OnResolve(dnssd_error_e result, dnssd_service_h service, void * userData)
 
 CHIP_ERROR ResolveAsync(chip::Dnssd::ResolveContext * rCtx)
 {
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    int ret;
+
+    assertChipStackLockedByCurrentThread();
+
     ChipLogDetail(DeviceLayer, "DNSsd %s", __func__);
 
-    int ret = dnssd_resolve_service(rCtx->mServiceHandle, OnResolve, rCtx);
+    if (rCtx->mInterfaceId == 0)
+    {
+        ret = dnssd_create_remote_service(rCtx->mType, rCtx->mName, nullptr, &rCtx->mServiceHandle);
+    }
+    else
+    {
+        char iface[IF_NAMESIZE + 1] = "";
+        VerifyOrExit(if_indextoname(rCtx->mInterfaceId, iface) != nullptr,
+                     ChipLogError(DeviceLayer, "if_indextoname() failed: %s", strerror(errno));
+                     err = CHIP_ERROR_POSIX(errno));
+        ret = dnssd_create_remote_service(rCtx->mType, rCtx->mName, iface, &rCtx->mServiceHandle);
+    }
+
+    VerifyOrExit(ret == DNSSD_ERROR_NONE,
+                 ChipLogError(DeviceLayer, "dnssd_create_remote_service() failed: %s", get_error_message(ret));
+                 err = TizenToChipError(ret));
+
+    ret = dnssd_resolve_service(rCtx->mServiceHandle, OnResolve, rCtx);
     VerifyOrReturnValue(ret == DNSSD_ERROR_NONE, TizenToChipError(ret),
                         ChipLogError(DeviceLayer, "dnssd_resolve_service() failed: %s", get_error_message(ret)));
 
     rCtx->mIsResolving = true;
-    return CHIP_NO_ERROR;
+exit:
```

Comment:
This combines VerifyOrExit and VerifyOrReturn. Please don't do that especially if you have no cleanup log.
Update the code to use return only.

Reply:
I'll update it